core/src/test/scala/ai/h2o/sparkling/TestUtils.scala

```diff
@@ -74,6 +75,29 @@ object TestUtils extends Matchers {
     producedNames shouldEqual expectedNames
   }
 
+  def assertDataFramesAreIdenticalWithSavingProducedToFile(
+      spark:SparkSession,
+      expected: DataFrame,
+      produced: DataFrame): Unit = {
+    val tmpDir = Files.createTempDirectory("dfComparison").toFile
+    val loadedProduced = try {
+      val path = tmpDir.getAbsolutePath + "/file"
+      produced.write.parquet(path)
+      spark.read.parquet(path)
+    } finally {
+      deleteOnExit(tmpDir)
+    }
+    assertDataFramesAreIdentical(expected, loadedProduced)
```

Comment:
I also check the implementation of this method and it is not actually testing the datasets are identical.

    scala> val left = Seq((1, 0.0), (1, 0.0), (1, 1.0))
    left: Seq[(Int, Double)] = List((1,0.0), (1,0.0), (1,1.0))

    scala> val right = Seq((1, 1.0), (1, 0.0), (1, 1.0))
    right: Seq[(Int, Double)] = List((1,1.0), (1,0.0), (1,1.0))

    scala> spark.createDataFrame(left).except(spark.createDataFrame(right)).count()
    res17: Long = 0                                                                 

    scala> spark.createDataFrame(right).except(spark.createDataFrame(left)).count()
    res18: Long = 0

    scala> spark.createDataFrame(right).distinct().count()
    res20: Long = 2

    scala> spark.createDataFrame(left).distinct().count()
    res21: Long = 2

The compared datasets are not identical because left has 2 zeros in second column, while the right has only 1.

Reply:
Nice catch, thanks a lot!